homeassistant/components/yessssms/notify.py

```diff
@@ -8,34 +8,49 @@
 
 from homeassistant.components.notify import PLATFORM_SCHEMA, BaseNotificationService
 
+from YesssSMS import YesssSMS
+
+from .const import CONF_PROVIDER
+
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
         vol.Required(CONF_RECIPIENT): cv.string,
+        vol.Optional(CONF_PROVIDER, default="YESSS"): cv.string,
     }
 )
 
 
 def get_service(hass, config, discovery_info=None):
     """Get the YesssSMS notification service."""
-    return YesssSMSNotificationService(
-        config[CONF_USERNAME], config[CONF_PASSWORD], config[CONF_RECIPIENT]
+
+    try:
+        yesss = YesssSMS(
+            config[CONF_USERNAME], config[CONF_PASSWORD], provider=config[CONF_PROVIDER]
+        )
+    except (KeyError, YesssSMS.UnsupportedProviderError) as ex:
```

Comment:
When can the `KeyError` happen?

Reply:
you are right, it shouldn't any more. I'm throwing `UnsupportedProviderError`.
will remove.